paddle/fluid/framework/ir/fc_mkldnn_pass.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/fc_mkldnn_pass.h"
+#include <string>
+#include <vector>
+#include "paddle/fluid/framework/eigen.h"
+#include "paddle/fluid/framework/lod_tensor.h"
+#include "paddle/fluid/platform/enforce.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+void Transpose(float* data, size_t rows, unsigned cols) {
+  std::vector<float> temp(rows * cols);
+
+  for (int i = 0; i < rows; i++) {
+    for (int j = 0; j < cols; j++) {
+      temp[j * rows + i] = data[i * cols + j];
+    }
+  }
+
+  memcpy(data, &temp[0], temp.size() * sizeof(float));
+}
```

Comment:
What's the reason for using std::vector here? If it's for automatic memory management, maybe you could just use std::unique_ptr<float[]> temp {new float[rows*cols]}? Or maybe use std::copy/std::copy_n instead of memcpy if you want to stick to std::vector?

Reply:
Does MKL have Transpose API?